catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -0,0 +1,700 @@
+                String pid = factoryPidDisabled;
+                if (autoActivateConfigurations && factoryPidMask.equals(bindingTypeToActivate)) {
+                    pid = factoryPid;
+                }
+                curConfig = configurationAdmin.createFactoryConfiguration(pid, null);
+            } else {
+                serviceConfigurationProperties.putAll(getConfigurationsFromDictionary(curConfig.getProperties()));
+            }
+
+            serviceConfigurationProperties.putAll(slotMap);
+            serviceConfigurationProperties.put(ID, configId);
+            serviceConfigurationProperties.put(SHORTNAME, configId);
+            serviceConfigurationProperties.put(RegistryObjectMetacardType.REGISTRY_ID, registryId);
+
+            curConfig.update(serviceConfigurationProperties);
+        }
+
+    }
+
+    private void deleteRegistryConfigurations(Metacard metacard)
+            throws IOException, InvalidSyntaxException {
+        String registryId = null;
+        Attribute registryIdAttribute =
+                metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID);
+        if (registryIdAttribute != null) {
+            registryId = registryIdAttribute.getValue()
+                    .toString();
+        }
+
+        Configuration[] configurations = configurationAdmin.listConfigurations(String.format(
+                CONFIGURATION_FILTER,
+                RegistryObjectMetacardType.REGISTRY_ID,
+                registryId));
+        if (configurations != null) {
+            for (Configuration configuration : configurations) {
+                configuration.delete();
+            }
+        }
+    }
+
+    /**
+     * Toggles a configuration between enabled and disabled.
+     *
+     * @param config The configuration to enable/disable
+     * @return The new enabled/disabled configuration
+     * @throws IOException
+     */
+    private Configuration toggleConfiguration(Configuration config) throws IOException {
+        String newFpid;
+        if (config.getFactoryPid()
+                .contains(DISABLED_CONFIGURATION_SUFFIX)) {
+            newFpid = config.getFactoryPid()
+                    .replace(DISABLED_CONFIGURATION_SUFFIX, "");
+        } else {
+            newFpid = config.getFactoryPid()
+                    .concat(DISABLED_CONFIGURATION_SUFFIX);
+        }
+        Dictionary<String, Object> properties = config.getProperties();
+        config.delete();
+        Configuration newConfig = configurationAdmin.createFactoryConfiguration(newFpid, null);
+        newConfig.update(properties);
+        return newConfig;
+    }
+
+    /**
+     * Returns the service binding slots as a map of string properties with the slot name as the key
+     *
+     * @param binding ServiceBindingType to generate the map from
+     * @return A map of service binding slots
+     */
+    private Map<String, Object> getServiceBindingProperties(ServiceBindingType binding) {
```

Comment:
The map that comes out of the converter @gordocanchola created is the whole service binding. This method is only interested in the slots and a couple of the attributes.